### PR TITLE
Fixed #29667 -- Prohibited whitespaces in URL path converters.

### PR DESCRIPTION
**Pull request by hramezani**
[ticket 29667](https://code.djangoproject.com/ticket/29667)

### CODE REVIEW

File: django/urls/resolvers.py [hunk 1]
```diff
@@ -206,6 +207,9 @@ def _route_to_regex(route, is_endpoint=False):
     For example, 'foo/<int:pk>' returns '^foo\\/(?P<pk>[0-9]+)'
     and {'pk': <django.urls.converters.IntConverter>}.
     """
+    if any((s in route) for s in string.whitespace):
```
Comment (carltongibson) [inline django/urls/resolvers.py]: You want to use set operations here: 

```suggestion
    if not set(route).isdisjoint(string.whitespace):
```

File: django/urls/resolvers.py [hunk 2]
```diff
@@ -206,6 +207,9 @@ def _route_to_regex(route, is_endpoint=False):
...
+        raise ImproperlyConfigured("URL route '%s' contain invalid character space." % route)
```
Comment (carltongibson) [inline django/urls/resolvers.py]:
```suggestion
        raise ImproperlyConfigured("URL route '%s' contains invalid whitespace." % route)
```

File: django/urls/resolvers.py [hunk 3]
```diff
@@ -206,6 +207,9 @@ def _route_to_regex(route, is_endpoint=False):
...
+
```
Comment (carltongibson) [inline django/urls/resolvers.py]: I don't think we need the blank line here.

File: tests/urlpatterns/tests.py [hunk 4]
```diff
@@ -130,6 +130,13 @@ def test_invalid_converter(self):
         with self.assertRaisesMessage(ImproperlyConfigured, msg):
             path('foo/<nonexistent:var>/', empty_view)
 
+    def test_space_in_route(self):
+        with self.assertRaisesMessage(
+            ImproperlyConfigured,
+            "URL route 'space/<int: num>' contain invalid character space."
```
Comment (carltongibson) [inline tests/urlpatterns/tests.py]: Can you format this like the case above? (With a `msg` variable, and then on a single line?)

File: django/urls/resolvers.py (hunk 2 above)
Comment (felixxm) [reply django/urls/resolvers.py]: Maybe without `invalid`? `"URL route '%s' contains whitespace."` or `"URL route '%s' cannot contain whitespace."`

Comment (carltongibson) [reply django/urls/resolvers.py]: Let's go with `...cannot contain whitespace.` (I briefly considered that, so it coming up twice, we'll take as a message from the universe.)

**Comment by hramezani** [comment]
@carltongibson Thanks for the review.
The patch is ready now.

**Pull request merged**